packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/knowledge/KnowledgeController.java

```diff
@@ -230,34 +230,59 @@ private void enrichModel(
 		}
 
 		// Update State Grounding
-		if (!model.get().isRegnet()) {
+		if (model.get().isPetrinet()) {
```

Comment:
`!model.get().isRegnet()` was purposful as we also have roughed in support for stock and flow, which isn't regnet and looks to go down this path?

Reply:
oh i see `isPetrinet()`is new. is stock and flow still relevant? our front end seems to think it is

Reply:
I fixed this up. But I noticed that skema also return "generic math models" on top of petrinets, and regnets.